doc/sphinxman/source/prog_debug_profile.rst

```diff
@@ -37,7 +37,23 @@ Debugging and Profiling
 Debugging
 ---------
 
-Instructions on running Psi4 with a debugger.
+The preferred method for debugging C++ code in Psi4 is with gdb. To run Psi4 in this way, you must specify the Python executable as the program. Arguments are provided using the ``--args`` flag. Here's an example.::
+  $~> gdb --args /usr/bin/python3 /path/to/psi4 input.dat
+
+In order to debug properly, Psi4 needs to be built with the ``CMAKE_BUILD_TYPE`` variable set to either ``Debug`` or ``RelWithDebInfo``. These will output debugging symbols that will allow gdb to print line numbers and
+inline function names.::
+  $~> cmake [your options here] -DCMAKE_BUILD_TYPE=Debug
+  $~> cmake [your options here] -DCMAKE_BUILD_TYPE=RelWithDebInfo
+
+Certain symbols may not be output even with this flag set. In general, any template function used should be visible, and anything with the ``PSI_API`` modifier will be visible. Other variables, functions, and classes will
+likely be hidden from the user. To make these symbols visible, you must modify a few variables. For an example, see `TiborGY's debug branch <https://github.com/psi4/psi4/compare/master...TiborGY:psi4:toc_dbg>`_.
+
+VSCode
+^^^^^^
+
+When using gdb wath VSCode, you should set the ``program`` entry to the Python executable, just as before. Arguments can then be placed in the ``args`` entry. If you are debugging a C++ plugin or backend code,
+the launch type should be ``cppdbg``.
+ 
```

Comment:
Thanks! While this is open, could you work in links to 
* https://github.com/psi4/psi4/blob/master/doc/sphinxman/source/build_planning.rst?plain=1
* https://github.com/psi4/PsiCon2020/blob/master/PsiCon2017/Turney-C%2B%2B.pdf

fwiw, my quick hack is to change all the CMakeCache `-O2` and `-O3` to `-O0` as well as `Release` to `Debug` since latter adds the `-g` and former removes optimizations